cmd/thanos/receive.go

```diff
@@ -521,9 +536,15 @@ func runReceive(
 				// Before quitting, ensure all blocks are uploaded.
 				defer func() {
 					<-uploadC
```

Comment:
Wow what's this ;p I did not know we have it. In such case why we don't merge those two goroutines into one? With above repeat. I don't think we want 2 syncs running in same time.